bundles/org.openhab.binding.bluetooth/src/test/java/org/openhab/binding/bluetooth/util/RetryFutureTest.java

```diff
@@ -151,10 +152,9 @@ void composeWithRetry1Cancel() throws InterruptedException {
             if (!latch.await(100, TimeUnit.MILLISECONDS)) {
                 fail("Timeout while waiting for latch");
             }
-            Thread.sleep(1);
             retryFuture.cancel(false);
 
-            assertTrue(composedFuture.isCancelled());
+            waitForAssert(() -> assertTrue(composedFuture.isCancelled()));
```

Comment:
The purpose of this test was to make sure that a composed future got cancelled when the retryFuture got cancelled. Since it possible that the retry future would get cancelled before the composed future was even created it was necessary to wait, or simply sleep, until we know that the composed future was created in another thread so we can attempt to cancel it.

The waitForAssert can still return false if the above race condition is not dealt with.

`retryFuture.cancel` should also cancel `composedFuture` before returning so adding any sort of `waitForAssert` is meaningless since it should have already occurred during the `retryFuture.cancel` call.